mllib/src/main/scala/org/apache/spark/mllib/stat/test/ChiSqTest.scala

```diff
@@ -109,7 +109,9 @@ private[stat] object ChiSqTest extends Logging {
           }
           i += 1
           distinctLabels += label
-          features.toArray.view.zipWithIndex.slice(startCol, endCol).map { case (feature, col) =>
+          val featureArray = features.toArray
+          (startCol until endCol).map { col =>
+            val feature = featureArray(col)
```

Comment:
Converting to Breeze should be an O(1) operation, using a reference copy, not a deep copy.  Breeze uses binary search on the indices, so it should be fairly efficient.  I think it's better than converting to a dense array.